scripts/aggregate-junit-xml.js

```diff
@@ -23,96 +23,166 @@
  * By default, all categories will be tested. To only process specific
  * categories, call the script with desired categories separated by a space.
  *
- * e.g. node aggregate-junit-xml.js integration vscode-integration
+ * e.g. 
+ * Return missing tests, but don't error: node aggregate-junit-xml.js -m -t integration vscode-integration
+ * Exit on error: node aggregate-junit-xml.js -m -t integration vscode-integration
  */
 
 const fs = require('fs-extra');
 const path = require('path');
 const shell = require('shelljs');
 
-const cwd = process.cwd();
-const packagesDir = path.join(cwd, 'packages');
-const aggregateDir = path.join(cwd, 'junit-aggregate');
+// Pass this param when you do not check for any crashed tests, but don't want to fail.
+// const RETURN_MISSING_PACKAGES_ARG = '-m';
+const TEST_TYPE_ARG = '-t';
+
+// const returnMissingPackages = process.argv.indexOf(RETURN_MISSING_PACKAGES_ARG) > - 1 ? false : true;
+
 const categoryToFile = {
   'vscode-integration': 'junit-custom-vscodeIntegrationTests.xml',
   integration: 'junit-custom-integrationTests.xml',
   unit: 'junit-custom-unitTests.xml',
   system: 'junit-custom.xml'
 };
 
-// process all test categories if none are specified in arguments
-let flags = new Set(process.argv.slice(2));
-if (flags.size === 0) {
-  flags = new Set(Object.keys(categoryToFile).map(c => `--${c}`));
-}
-
-if (!fs.existsSync(aggregateDir)) {
-  shell.mkdir(path.join(cwd, 'junit-aggregate'));
-}
-
 const missingResults = {
   'vscode-integration': [],
   integration: [],
   unit: [],
   system: []
 };
 
-for (const entry of fs.readdirSync(packagesDir)) {
-  const packagePath = path.join(packagesDir, entry);
+function getTestCategories() {
+  let flags;
+  if (process.argv.indexOf(TEST_TYPE_ARG) > -1) {
```

Comment:
Explaining why I'm keeping this:

Originally I needed to make this change because I had another flag that I was using as well. This other flag was going to be used to determine when we wanted the script to return the failed package list, but not throw an exit code. Originally I was going to relaunch the tests outside of this script.

I opted to go with the simpler approach of just relaunching the tests from within this script itself.